mix.exs

```diff
@@ -30,18 +30,10 @@ defmodule Tesla.Mixfile do
               "Documentation" => "http://hexdocs.pm/tesla/0.1.0/"}]
   end
 
-  defp deps(:dev) do
-    [{:ibrowse, github: "cmullaparthi/ibrowse", tag: "v4.1.1"},
+  defp deps do
+    [{:ibrowse, github: "cmullaparthi/ibrowse", tag: "v4.1.1", optional: true},
```

Comment:
`optional: true` sets the dependency is optional and used only to specify requirements. It is left to the user to include the dependency only if they wish to use it.

This is the same way dependencies are setup in `Plug` and `Ecto`. `Ecto` has an optional dependency on `postgrex` and `mariaex` and it is required for the user to include the dependency they wish to use.

I think `exjsx` should eventually not be a dependency at all. A JSON encoder/decoder should be left to the user to include and specify when adding the middleware (Similar to how `Plug` allows the JSON Parser to be configured).